src/postgres_mcp/server.py

```diff
@@ -14,373 +12,257 @@
-                result_text += str(slow_queries)
-                return [types.TextContent(type="text", text=result_text)]
-            else:
-                message = (
-                    f"The '{PG_STAT_STATEMENTS}' extension is required to report slow queries, but it is not currently installed.\n\n"
-                    f"You can ask me to install it using the 'install_extension_pg_stat_statements' tool.\n\n"
-                    f"**Is it safe?** Installing '{PG_STAT_STATEMENTS}' is generally safe and a standard practice for performance monitoring. It adds performance overhead by tracking statistics, but this is usually negligible unless your server is under extreme load. It requires database privileges (often superuser) to install.\n\n"
-                    f"**What does it do?** It records statistics (like execution time, number of calls, rows returned) for every query executed against the database.\n\n"
-                    f"**How to undo?** If you later decide to remove it, you can ask me to run 'DROP EXTENSION {PG_STAT_STATEMENTS};'."
-                )
-                return [types.TextContent(type="text", text=message)]
 
-        except Exception as e:
-            print(
-                f"Error checking or querying {PG_STAT_STATEMENTS}: {e}", file=sys.stderr
-            )
-            raise
+@mcp.resource("postgres://resources")
+@handle_db_errors
+async def list_resources() -> list[types.Resource]:
+    """List available database tables as resources."""
+    sql_driver = get_safe_sql_driver()
+    rows = sql_driver.execute_query(
+        "SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'"
+    )
+    tables = [row.cells["table_name"] for row in rows] if rows else []
+
+    base_url = "postgres://"
+
+    resources = [
+        types.Resource(
+            uri=AnyUrl(f"{base_url}/{table}/schema"),
+            name=f'"{table}" database schema',
+            description=f"Schema for table {table}",
+            mimeType="application/json",
+        )
+        for table in tables
+    ]
 
+    extensions_uri = AnyUrl(f"{base_url}/_extensions")
+    resources.append(
+        types.Resource(
+            uri=extensions_uri,
+            name="Installed PostgreSQL Extensions",
+            description="List of installed PostgreSQL extensions in the current database.",
+            mimeType="application/json",
+        )
+    )
+
+    return resources
+
+
+@mcp.resource("postgres://_extensions")
+@handle_db_errors
+async def extensions_resource() -> str:
+    """Get information about installed PostgreSQL extensions."""
+    sql_driver = get_safe_sql_driver()
+    rows = sql_driver.execute_query(
+        "SELECT extname, extversion FROM pg_extension ORDER BY extname;"
+    )
+    extensions = [row.cells for row in rows] if rows else []
+    return str(extensions)
+
+
+@mcp.resource("postgres://{table_name}/schema")
```

Comment:
Should we be putting a *name* and *description* on these resources?

Reply:
The name is taken from the function name. I can add a description.

Reply:
done